src/language/ast.ts

```diff
@@ -63,15 +82,99 @@ export interface ASTKindToNode {
   ModelDescription: ModelDescriptionNode;
   Name: NameNode;
   OptionalName: OptionalNameNode;
+  // Endpoints types
+  EndpointNamespaceTypeDefinition: EndpointNamespaceTypeDefinitionNode;
+  EndpointVerb: EndpointVerbNode;
+  EndpointUrl: EndpointUrlNode;
+  EndpointTypeDefinition: EndpointTypeDefinitionNode;
+  EndpointResponse: EndpointResponseNode;
+  EndpointSecurityDefinition: EndpointSecurityDefinitionNode;
 }
 
 export interface DocumentNode {
   readonly kind: 'Document';
   readonly loc?: Location;
-  readonly definitions: ReadonlyArray<TypeDefinitionNode>;
+  readonly definitions: ReadonlyArray<TypeDefinitionNode | EndpointNamespaceTypeDefinitionNode>;
+}
+
+export type TypeDefinitionNode =
+  | ModelTypeDefinitionNode
+  | EnumTypeDefinitionNode
+
+export interface EndpointNamespaceTypeDefinitionNode {
+  readonly kind: 'EndpointNamespaceTypeDefinition';
+  readonly tag?: NameNode;
+  readonly description?: ModelDescriptionNode;
+  readonly endpoints?: ReadonlyArray<EndpointTypeDefinitionNode>;
```

Comment:
it's better to have not an optional, but empty array

Reply:
Briefly about hidden classes and what I mean when I talk about monomorphism

https://youtu.be/ww7LBKx1CEk